File: quiz/src/main/java/com/quiz/quiz/entities/formysql/Answer.java
package com.quiz.quiz.entities.formysql;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "answers")
@Data
public class Answer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String value;

    @Column(name = "is_correct")
    private Boolean isCorrect = false;

    @ManyToOne
    @JoinColumn(name = "question_id")
    private Question question;
}

File: backend/quiz-creation-service/domains/domains/Answer.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace domains.domains
{
    [Table("answers")]
    public class Answer
    {
        [Column("id")]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [Column("text")]
        public string Text { get; set; } = null!;

        [Column("is_correct")]
        public bool IsCorrect { get; set; }

        [Column("question_id")]
        public int QuestionId { get; set; }

        [ForeignKey("QuestionId")]
        public Question Question { get; set; } = null!;
    }
}
